lib/utils/job-utils/HTTP-tool.js

```diff
@@ -0,0 +1,195 @@
+//Copyright 2016, EMC, Inc.
+
+/*jshint node: true*/
+/*jshint multistr: true*/ 
+
+'use strict';
+
+var di = require('di');
+
+module.exports = HttpFactory;
+
+di.annotate(HttpFactory, new di.Provide('JobUtils.HttpTool'));
+di.annotate(HttpFactory, new di.Inject(
+    'Assert',
+    'Promise',
+    '_'
+));
+
+function HttpFactory(assert, Promise, _) {
+    
+    /**
+    * Tool class that does HTTP methods
+    * @constructor
+    */
+    function HttpTool(){
+        this.settings = {};
+        
+        // ********** Helper Functions/Members **********
+        
+        /**
+        * Make sure that the method is a valid HTTP method
+        * @param {string} - testMethod: the intended method that for 
+        * http request.
+        * @return {boolean} - Whether the testmethod is valid
+        */
+        var validMethod = {'GET': true, 'PUT': true, 'POST':true, 
```

Comment:
I need a datastructure similar to "HashSet" in Java (with time complexity O(1) vs O(n)). And I find out that similar javascript practice of doing that is using a object and put them as keys.